### ISSUE TITLE
Kubeflow 1.2 release doc changes

### ISSUE THREAD

**Issue by Jeffwan** [kubeflow/website#2322]
I create this issue to track coming 1.2 doc changes.  WG owners and platform owners, it would be great to link your WIG or PRs there to help on tracking. 

We will cut a 1.2 branch from masters around Nov 8.

Docs for versioning the website: https://github.com/kubeflow/kubeflow/blob/master/docs_dev/releasing.md#version-the-website

**Comment by Jeffwan** [comment]
@8bitmp3 Do you have capacity to drive 1.2 website release?

**Comment by andreyvelich** [comment]
@Jeffwan From AutoML WG these PRs are related to 1.2:

- https://github.com/kubeflow/website/pull/2312
- https://github.com/kubeflow/website/pull/2315
- https://github.com/kubeflow/website/pull/2316
- https://github.com/kubeflow/website/pull/2320
- https://github.com/kubeflow/website/pull/2321
- https://github.com/kubeflow/website/pull/2324

That is the main issue to track the release: https://github.com/kubeflow/katib/issues/1360.

/cc @johnugeorge @gaocegege

**Comment by RFMVasconcelos** [comment]
Thank you for creating this issue @Jeffwan! 

I can help drive this effort. Do you say you'll cut a 1.1 branch on Nov 8th?

**Comment by RFMVasconcelos** [comment]
Hi @Jeffwan @andreyvelich !
 
So the way we do it is that we change `master` to be the latest version, now -> `v1.2`, and we create a new branch for the previous version - `v1.1`.

If you look, today we don't have a `v1.1-branch` on the website repo as `v1.1` is now in `master`

**Comment by RFMVasconcelos** [comment]
Happy to drive the cut of `v1.1-branch` this week.

**Comment by RFMVasconcelos** [comment]
@jlewi, it seems that I do not have the permissions to create a new branch upstream. How should I proceed?

**Comment by jlewi** [comment]
@RFMVasconcelos 
https://github.com/kubeflow/community/blob/master/how-to/github_admin.md
https://github.com/kubeflow/internal-acls/blob/4535a60993946dcb57a05cd33552518030410ac5/github-orgs/kubeflow/org.yaml#L1100
https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization

**Comment by RFMVasconcelos** [comment]
Thank you!

**Comment by RFMVasconcelos** [comment]
@jlewi I created the PR to update permissions. ☝️

**Comment by Jeffwan** [comment]
@RFMVasconcelos I get some feedbacks from WG owners. Most WGs owners just finish code and manifest updates this week. They are working on docs changes.  Can we postpone the website branch cut a little bit late? I propose to do it on Nov 11 instead of the end of this week. What do you think? 

one example is https://github.com/kubeflow/website/issues/2322#issuecomment-719094241 /cc @andreyvelich

**Comment by RFMVasconcelos** [comment]
@Jeffwan that's ok! I can do it early next week ;) 

These updates are things we will want to stay in Kubeflow v1.1 docs?

**Comment by Jeffwan** [comment]
em. See https://github.com/kubeflow/kubeflow/issues/5371#issuecomment-720488097 
doc repo is managed in a different way. it lags 1 version. The time we cut v1.1-branch, that means v1.1 is not the latest version and all changes on master will also go online (kubeflow.org points to master).  

So I think we don't want those changes coming into v1.1 since they are v1.2 only. Can we accept new changes on master and then cut branch from a stable commit without v1.2 related changes? Any better ideas?

**Comment by andreyvelich** [comment]
I think if we want to cut the v1.1 website release, we need to choose one of the past commits and make the release. For example: https://github.com/kubeflow/website/commit/97dc74694ac0b45c05e895892f2676d8c8412328.
I believe the multi-user doc for the Kubeflow Pipelines was ready at this time.

If we can't find the commit for the v1.1 website, we should skip this version and directly cut v1.2 for the Kubeflow 1.2 release, once all required PRs docs will be merged to the master.

As I can see, currently the Kubeflow website points the master version to the 1.1 branch:
![Screenshot 2020-11-06 at 22 36 16](https://user-images.githubusercontent.com/31112157/98421143-9c975c00-2080-11eb-8e2f-f22180985699.png)

I think it is not correct, because:
1. v1.1-branch doesn't exists (https://github.com/kubeflow/website/branches/all?page=1).
2. Master version of the website should always point to the master branch.

What do you think @Jeffwan @RFMVasconcelos ?

/cc @Bobgy @jlewi

**Comment by Jeffwan** [comment]
@andreyvelich  It's by design. see https://github.com/kubeflow/kubeflow/issues/5371#issuecomment-720488097

 it's a little bit different from other projects. I think both solution work. website normally is changed frequently when user notice there's issue for latest branch. If we follow kubeflow/kubeflow way, then 

1. There will be lots of cherry-picks or user directly submit PR against v1.1-branch
2. Kubeflow.org needs to switch to last stable version like v1.1 instead of master.  

The v1.1 in banner doesn't mean there's a tag or version named v1.1. It's same as master. I know this is confuse, I think if most people want to do it in a different way, that works as well.

**Comment by jlewi** [comment]
@Jeffwan is correct

* We wait to cut the v1.1 docs branch because docs lag behind 
* We cut the v1.1 branch only when we are ready to start updating the docs for 1.2
   * We are probably at a point where people want to start committing 1.2 changes so cutting the 1.1 branch is a good idea
   * As @andreyvelich mentions we can cut the branch on a past commit if there have already been changes that are only relevant to 1.2

* We can independently control whether www.kubeflow.org points to master or a 1.X branch
* So after cutting 1.1 docs branch we should update www.kubeflow.org to point at 1.1
* We can then develop 1.2 docs on master which will be available at master.kubeflow.org
* When we are ready to release 1.2 we should flip www.kubeflow.org to point to the master branch containing 1.2 docs.

**Comment by RFMVasconcelos** [comment]
Sounds good @jlewi!

I am happy to make the `v1.1-branch` cut + redirect when we're ready. 

@andreyvelich are there any merged docs that are v1.2 specific hence forcing a cut from a past commit? If not I'd rather do the cut at the current state.

The commit you mention is 1 month old, which means some relevant commits might be lost, including some of our recent efforts to remove "outdated" banners. 

WDYT?

**Comment by andreyvelich** [comment]
@RFMVasconcelos From AutoML WG only this commit is related to 1.2: https://github.com/kubeflow/website/commit/1df2cb955df7213ce30e997b4801fd15741dc4a9.

**Comment by Jeffwan** [comment]
@RFMVasconcelos 
![image](https://user-images.githubusercontent.com/4739316/98706479-bd3b0c80-2333-11eb-9bf6-6c5fd044538c.png)
I think this should be good? 

I am not sure about #2327 that manifest change is not in 1.1-branch but master

**Comment by RFMVasconcelos** [comment]
Ok @Jeffwan! 

From the meeting, it seemed like nobody had big objections. I will use that commit and then ask the leads to review if there is any inconsistency on the `v1.1-branch`

**Comment by Jeffwan** [comment]
Great! @RFMVasconcelos

**Comment by RFMVasconcelos** [comment]
Branch created -> https://github.com/kubeflow/website/tree/v1.1-branch

**Comment by RFMVasconcelos** [comment]
Netlify deployed v1.1 version:
https://v1-1-branch--competent-brattain-de2d6d.netlify.app/

**Comment by RFMVasconcelos** [comment]
Hey @jlewi @Bobgy, I've done steps 1 & 2 of the ["Creating and publishing a website branch for vX.Y" guide](https://github.com/kubeflow/kubeflow/blob/master/docs_dev/releasing.md#creating-and-publishing-a-website-branch-for-vxy). ☝️ 

I'm stuck on step 3, as I don't have access to the [Cloud DNS of Kubeflow.org](https://console.cloud.google.com/net-services/dns/api/enable?nextPath=%2Fzones&project=kubeflow-dns&organizationId=714441643818).

How should I proceed?

**Comment by jlewi** [comment]
One of @kubeflow/google-admins will have to do the DNS steps.

**Comment by RFMVasconcelos** [comment]
Thank you for clarifying @jlewi. Could any of the @kubeflow/google-admins provide a timeline for this? 

Many thanks!

**Comment by Bobgy** [comment]
I'll take a look tomorrow

**Comment by Bobgy** [comment]
I've updated DNS record, https://v1-1-branch.kubeflow.org/ shows up correctly. It hasn't updated SSL certificate yet, I'm contacting netlify to set it up.

**Comment by Bobgy** [comment]
After SSL ready, I'll do "Changing the version www.kubeflow.org points to" step

**Comment by RFMVasconcelos** [comment]
Awesome! Let me know if you want me to do any of the next steps

**Comment by Bobgy** [comment]
OK, SSL is now ready. I've also switched netlify to build v1.1-branch as kubeflow.org content.
@RFMVasconcelos can you continue to update master branch's label so it shows as v1.2

**Comment by RFMVasconcelos** [comment]
Hi @Bobgy! 

Thank you for the help! I can take care of that until EOW

**Comment by Jeffwan** [comment]
@RFMVasconcelos We cut code release. do you want to update the docs?

**Comment by RFMVasconcelos** [comment]
Hey @Jeffwan! 

I have updated the website versioning -> #2365 

Also opened issues for the following steps that are in the [versioning guide](https://github.com/kubeflow/kubeflow/blob/master/docs_dev/releasing.md#create-website-branch):
* #2366 for updating shortcodes for config files and KFDefs
* #2367 for K8s version matrix

**Comment by thesuperzapper** [comment]
@RFMVasconcelos are you going to cut a 1.2 branch, rather than just pointing to master?

**Comment by Bobgy** [comment]
Shall we resolve the urgent issue first? We need to decide on a date when master will be shown as default for www.kubeflow.org instead of v1.1-branch.kubeflow.org.

Doc updates on GCP side is almost done, I have one last PR waiting to be merged.

**Comment by Bobgy** [comment]
/cc @Jeffwan 
maybe you should make the call for "when we show 1.2 docs by default".

**Comment by RFMVasconcelos** [comment]
@thesuperzapper we're going to point to master. 

It seems that we're only missing [this step](https://github.com/kubeflow/kubeflow/blob/master/docs_dev/releasing.md#changing-the-version-wwwkubefloworg-points-to) to update to v1.2. 

I think we're ready to do this @Bobgy. I do not have access to Netlify, perhaps this needs to be done by a Googler?

**Comment by Bobgy** [comment]
GCP side is ready.

/cc @kubeflow/aws @kubeflow/arrikto @kubeflow/azure @Jeffwan @animeshsingh 

Did I miss any stakeholders?

If no one raises an objection, I will switch doc to master next Tuesday.

**Comment by 8bitmp3** [comment]
@Bobgy Double check with Seldon @cliveseldon and KFServing @andreyvelich 

@animeshsingh Does IBM have a `@kubeflow/ibm` team handle? @Tomcli

**Comment by 8bitmp3** [comment]
cc @kubeflow/wg-automl-leads @kubeflow/wg-notebook-leads @kubeflow/wg-pipeline-leads @kubeflow/wg-training-leads @kubeflow/wg-serving-leads just to keep everyone in the loop

**Comment by andreyvelich** [comment]
From AutoML WG all required docs were updated on November 14th.
That is the latest commit: https://github.com/kubeflow/website/commit/b2ad541ebc0e03e7c1e8c78f969d4d33fff7800b.

**Comment by Bobgy** [comment]
Updated default branch to master, once the next commit lands in master, the website will switch over to master as default.

**Comment by PatrickXYS** [comment]
Update from AWS Side, we have updated our doc to v1.2

**Comment by RFMVasconcelos** [comment]
It seems that v1.2 docs are live!! 🚀 

Thank you for the help @Bobgy! 

/close
